apps/rig_tests/test/proxy/response_from/http_test.exs

```diff
@@ -68,4 +74,76 @@ defmodule RigTests.Proxy.ResponseFrom.HttpTest do
     # ...the response received is exactly what the service mock returned:
     assert Jason.decode!(res_body) == sync_response
   end
+
+  test_with_server "Given response_from is set to http_internal, the http response is taken from the internal HTTP endpoint." do
+    test_name = "proxy-http-response-from-http-internal"
+
+    api_id = "mock-#{test_name}-api"
+    endpoint_id = "mock-#{test_name}-endpoint"
+    endpoint_path = "/#{endpoint_id}"
+    sync_response = %{"this response" => "the client never sees this response"}
+    async_response = %{"message" => "this is the async response that reaches the client instead"}
+
+    route(endpoint_path, fn %{query: %{"correlation" => correlation_id}} ->
+      event =
+        Jason.encode!(%{
+          specversion: "0.2",
+          type: "rig.async-response",
+          source: "fake-service",
+          id: "1",
+          rig: %{correlation: correlation_id},
+          data: async_response
```

Comment:
I think I'd do the async response outside the request handler, which would be the realistic use case and also make sure we're not simply forwarding the first response we got, but actually ignore the sync response

Reply:
Can you elaborate how would you do it? Since, if you won't do the response in this handler it won't get pass `%HTTPoison.Response{status_code: res_status, body: res_body} = HTTPoison.get!(request_url)` due to timeout.

Reply:
Indeed, that didn't make any sense ^_^ please ignore my comment.